src/pykeen/training/training_loop.py

```diff
@@ -1080,6 +1088,18 @@ def _save_state(
         else:
             lr_scheduler_state_dict = self.lr_scheduler.state_dict()
 
+        # TODO: Find easy and robust way to determine whether this is not a PyKEEN provided dataset
```

Comment:
@cthoyt 
Please have a look at the TODO.
Also, we should decide whether it would be useful to save those mappings for all datasets, since this would standardize our API in the way that applications can count on being able to retrieve the entity_to_id and relation_to_id mappings from the checkpoint.

Reply:
I would support storing them also for pykeen datasets (and maybe add a flag with which you can disable this if you really want to).

Reply:
@mberr I have now changed the code to safe them as default in https://github.com/pykeen/pykeen/pull/498/commits/1db07924d72ec12d960c027590d45df49b924f3c.
I have thought about the flag, but Imo it doesn't add real value for the following reasons

- It's one more flag and we slowly start drowning in flags
- It has no impact on the KGEM as such
- Even for large datasets like YAGO310, with over 123,000 entities, the entity mapping dict is not even 3MB big, this compared to the model size is negligible, especially considering that not saving the mapping could render the model useless with custom datasets
- It will make the sanity check, i.e. making sure the original model embeddings are aligned with the current triples factory, the standard, which is a very reasonable thing to have Imo :)

What do you think?

Reply:
I agree with laurent - let's do it by default so there's no messing around.